### PR TITLE
Mistral API docs & test using Mistral client

### PR DESCRIPTION
**Pull request by lgrammel**
## Summary
* Add next/Mistral example
* Add Mistral docs page
* Add Mistral test

### CODE REVIEW

File: docs/pages/docs/guides/providers/mistral.mdx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  const response = await client.chatStream({
+    model: 'mistral-tiny',
+    stream: true,
+    max_tokens: 1000,
+    messages,
+  });
+
+  // Convert the response into a friendly text-stream. The Mistral client responses are
+  // compatible with the Vercel AI SDK OpenAIStream adapter.
+  const stream = OpenAIStream(response);
```
Comment (jaredpalmer) [inline docs/pages/docs/guides/providers/mistral.mdx:63]: Is this correct? It is compatible with an OpenAIStream?

File: docs/pages/docs/guides/providers/mistral.mdx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+// Note: There are no types for the Mistral API client yet.
+// @ts-ignore
+import MistralClient from '@mistralai/mistralai';
+
+export async function POST(req: Request) {
+  // Extract the `prompt` from the body of the request
+  const { prompt } = await req.json();
+
+  const client = new MistralClient(process.env.MISTRAL_API_KEY || '');
```
Comment (jaredpalmer) [inline docs/pages/docs/guides/providers/mistral.mdx:137]: We should likely instantiate the client outside of the handler?

File: docs/pages/docs/guides/providers/mistral.mdx [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+```
+
+### Wire up the UI
+
+We can use the [`useCompletion`](/docs/api-reference/use-completion) hook to make it easy to wire up the UI. By default, the `useCompletion` hook will use the `POST` Route Handler we created above (it defaults to `/api/completion`). You can override this by passing a `api` prop to `useCompletion({ api: '...'})`.
+
+```tsx filename="app/page.tsx" showLineNumbers
+'use client';
+import { useCompletion } from 'ai/react';
+export default function Completion() {
```
Comment (jaredpalmer) [inline docs/pages/docs/guides/providers/mistral.mdx:163]: Add a new line between the imports and components

File: examples/next-mistral/app/api/chat/route.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+// Note: There are no types for the Mistral API client yet.
+// @ts-ignore
+import MistralClient from '@mistralai/mistralai';
+
+export async function POST(req: Request) {
+  // Extract the `messages` from the body of the request
+  const { messages } = await req.json();
+
+  const client = new MistralClient(process.env.MISTRAL_API_KEY || '');
```
Comment (jaredpalmer) [inline examples/next-mistral/app/api/chat/route.ts:11]: Move client instantiation outside of the handler?

**Review by jaredpalmer — CHANGES_REQUESTED** [review]

File: docs/pages/docs/guides/providers/mistral.mdx (hunk 1 above)
Comment (lgrammel) [reply docs/pages/docs/guides/providers/mistral.mdx:63]: yes double checked and also ran the code

**Review by jaredpalmer — APPROVED** [review]

**Pull request merged**